server/src/main/scala/actors/ShiftsActorBase.scala

```diff
@@ -115,6 +135,13 @@ class ShiftsActorBase(now: () => SDateLike) extends RecoveryActorLike with Persi
     case u =>
       log.info(s"unhandled message: $u")
   }
+
+  def applyUpdatedShifts(existingAssignments: Seq[StaffAssignment], shiftsToUpdate: Seq[StaffAssignment]): Seq[StaffAssignment] = shiftsToUpdate
+    .foldLeft(existingAssignments) {
+      case (assignmentsSoFar, updatedAssignment) =>
+        assignmentsSoFar.filterNot(existing =>
+          existing.startDt == updatedAssignment.startDt && existing.terminalName == updatedAssignment.terminalName)
```

Comment:
Is there an id or created at field to match on? Unless we know for sure this is the correct staff assignment by the start date time and terminal name.

Reply:
This is essentially the unique key due to our interface only allowing one 'shift assignment' per 15 minutes per terminal